### PR TITLE
[Example] Add cg_possion.py, a Poisson solver based on conjugate gradients

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
video
<del>I found this example running extremely slow on `ti.opengl`, does this means we also need BLS to accel programs like this?</del>Didn't merge the TLS PR.

### CODE REVIEW

File: examples/cg_possion.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import taichi as ti
+import math
+ti.init(ti.cuda)
```
Comment (yuanming-hu) [inline examples/cg_possion.py:3]:
```suggestion
ti.init(arch=ti.cuda)
```

File: examples/cg_possion.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    for i in ti.grouped(x):
+        dAd += d[i] * A(d, i)
+    for i in ti.grouped(x):
+        alpha += r[i]**2 / dAd
+    for i in ti.grouped(x):
+        x[i] = x[i] + alpha * d[i]
+        r[i] = r[i] - alpha * A(d, i)
+        beta += r[i]**2 / (alpha * dAd)
+    for i in ti.grouped(x):
+        d[i] = r[i] + beta * d[i]
```
Comment (yuanming-hu) [inline examples/cg_possion.py:46]: Use `I` instead of `i` for packed indices?

File: examples/cg_possion.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+@ti.kernel
+def init():
+    for i in ti.grouped(x):
+        d[i] = b[i] - A(x, i)
+        r[i] = d[i]
+
+
+@ti.kernel
+def substep():
+    alpha, beta, dAd = eps, eps, eps
```
Comment (yuanming-hu) [inline examples/cg_possion.py:36]: `beta` doesn't have to start with `eps` initially. Please initialize these to zero and add `eps` only when they are used as denominators.

**Comment by archibate** [comment]
> Merging #1595 into master will increase coverage by 22.41%.

What happened to codecov these days... Seems it's Travis build taking too long time therefore failed to upload coverage information? We'd better switch to GHCI quicker.

**Pull request merged**